core/services/relay/evm/functions.go

```diff
@@ -36,35 +43,76 @@ func (p *functionsProvider) ContractTransmitter() types.ContractTransmitter {
 	return p.contractTransmitter
 }
 
-func NewFunctionsProvider(chainSet evm.ChainSet, rargs relaytypes.RelayArgs, pargs relaytypes.PluginArgs, lggr logger.Logger, ethKeystore keystore.Eth, pluginType functionsRelay.FunctionsPluginType) (relaytypes.Plugin, error) {
-	configWatcher, err := newFunctionsConfigProvider(pluginType, chainSet, rargs, lggr)
+func (p *functionsProvider) LogPollerWrapper() evmRelayTypes.LogPollerWrapper {
+	return nil
+}
+
+func (p *functionsProvider) Start(ctx context.Context) error {
+	return p.StartOnce("FunctionsProvider", func() error {
+		if err := p.configWatcher.Start(ctx); err != nil {
```

Comment:
I think I understand needing to start the new logPollerWrapper, but I don't userstand why we now also have to start the configWatcher when we did not have to do this previously.

Reply:
We did. When configWatcher was embedded in the provider struct, this Start() method was the watcher's Start() method. I.e. the Watcher was being started directly and now we're starting it via provider's Start() so we can plug in other stuff when needed.

Reply:
That makes sense. Thanks